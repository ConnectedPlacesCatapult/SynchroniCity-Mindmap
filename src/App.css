html, body {
  margin: 0;
  padding: 0;
  scroll-behavior: smooth;
}

.App {
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  background-image: url('./images/bg.jpg');
  background-size: cover;
  color: white;
  overflow: hidden;
}

h1 {
  font-weight: 400;
  font-size: 2em;
}

p {
  font-weight: 300;
  font-size: 1em;
}

.center {
  text-align: center;
}

#resultsSection {
  /* position: absolute; */
  padding-top: 12.5vh;
  height: 100vh;
}

#topicsSection {
  /* position: absolute; */
  padding-top: 25vh;
  height: 100vh;
}