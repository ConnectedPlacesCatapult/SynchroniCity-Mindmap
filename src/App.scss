body, html {
  margin: 0 auto;
  padding: 0;
}

h1 {
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 2em;
}