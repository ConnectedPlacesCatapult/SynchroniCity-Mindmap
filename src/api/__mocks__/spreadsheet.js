const fakeTag0Unique = [
    { 0: "Internet of Things",
      1: "Public Realm",
      2: "MIMs",
      3: "Mobility",
      4: "Health",
      5: "Smart City Governance",
      6: "Smart City ICT",
      7: "Data Management & AI",
      8: "Buildings",
      9: "Citizen Centric Services",
      10: "Environment",
      11: "Utilities"},
];

const fakeFilteredStandards = [
    {
        serial: "ETSI TS 103 410-4 V1.1.1 (2019-05)",
        url: "https://www.etsi.org/deliver/etsi_ts/103400_103499/10341004/01.01.01_60/ts_10341004v010101p.pdf",
        title_full: "SmartM2M; Extension to SAREF; Part 4: Smart Cities Domain",
        abstract: "The present document is a technical specification of SAREF4Cities, an extension of SAREF for the Smart Cities domain. This extension has been created by investigating resources from potential stakeholders of the ontology, such as standardization bodies (e.g. Open Geospatial Consortium), associations (e.g. Spanish Federation of Municipalities and Provinces), IoT platforms (e.g. FIWARE) and European projects and initiatives (e.g. ISA2  programme) as reported in ETSI TR 103 506",
        date_published: "28/09/2018",
        tag0: "MIMs",
        tag1: "Semantic Data Model",
        tag2: "Ontology",
        tag3: "Smart Health",
        tag4: "Street Lighting",
        tag5: "Parking",
        tag6: "Air Quality",
        tag7: "Mobility",
        organisation: "ETSI TC SmartM2M",
        type: "Specification",
        ICS: ""},
    {
        serial: "ETSI GS CIM 009 V1.1.1 (2019-01)",
        url: "http://webapp.etsi.org/workprogram/Report_WorkItem.asp?WKI_ID=54473",
        title_full: "Context Information Management (CIM); NGSI-LD API",
        abstract: "This Group Specification provides additions and corrections to the GS-004 prelimAPI specification, based on feedback about GS-004 received from other SDOs as well as developers in the linked-data, internet-of-things, and mobile-apps and smart-applications communities, as well as from end users and stakeholders. In particular open issues and proposed features in Annexes of the GS-004 document, including the topics of privacy/security, will be considered for inclusion as part of this specification.",
        date_published: "22/01/2019",
        tag0: "MIMs",
        tag1: "API",
        tag2: "Context Information Management",
        tag3: "",
        tag4: "",
        tag5: "",
        tag6: "",
        tag7: "",
        organisation: "ETSI ISG CIM",
        type: "Specification",
        ICS: ""}
]

const fakeFilteredSubStandards = [
    {
        serial: "ETSI GS CIM 009 V1.1.1 (2019-01)",
        url: "http://webapp.etsi.org/workprogram/Report_WorkItem.asp?WKI_ID=54473",
        title_full: "Context Information Management (CIM); NGSI-LD API",
        abstract: "This Group Specification provides additions and corrections to the GS-004 prelimAPI specification, based on feedback about GS-004 received from other SDOs as well as developers in the linked-data, internet-of-things, and mobile-apps and smart-applications communities, as well as from end users and stakeholders. In particular open issues and proposed features in Annexes of the GS-004 document, including the topics of privacy/security, will be considered for inclusion as part of this specification.",
        date_published: "22/01/2019",
        tag0: "MIMs",
        tag1: "API",
        tag2: "Context Information Management",
        tag3: "",
        tag4: "",
        tag5: "",
        tag6: "",
        tag7: "",
        organisation: "ETSI ISG CIM",
        type: "Specification",
        ICS: ""
    },
]

export const getTag0 = () => {
    return new Promise((resolve, reject) => {
        resolve(fakeTag0Unique);
    });
};

export const getTags = () => {
    return new Promise((resolve, reject) => {
        resolve(fakeFilteredStandards);
    });
};

export const getFilteredSubStandards = () => {
    return new Promise((resolve, reject) => {
        resolve(fakeFilteredSubStandards);
    });
};

