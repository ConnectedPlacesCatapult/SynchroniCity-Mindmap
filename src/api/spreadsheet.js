import Tabletop from 'tabletop';

export var rows;

export const getTag0 = () => {
    return new Promise((resolve, reject) => {
        Tabletop.init({
            key: '1nogRbK3sIbF0T2bUPar9YLqvKPUWwFaSk5jk_hGBq-g',
            callback: googleData => {
                // all data stored in data
                var data = googleData['standards-datastore'];
                // console.log('data --->', data);

                // rows stored in rows
                rows = data['elements'];
                var tag0Array = []
                for (let step = 0; step < Object.keys(rows).length; step++) {
                    var tag0 = rows[step]['tag0'];
                    tag0Array.push(tag0)
                    }
                var tag0Unique = [];
                resolve(tag0Unique = Array.from(new Set(tag0Array)));
                // console.log('tag0 --->', tag0Unique);
            },
        });
    });
}


// export const getTags = (rows, tag0) => {
//     return new Promise((resolve, reject) => {
//         var tagAll = []
//         var filteredStandards = []
//         for (let step = 0; step < Object.keys(rows).length; step++) {
//             if (rows[step]['tag0'] === tag0) {
//                 tagAll.push(rows[step]['tag1']);
//                 tagAll.push(rows[step]['tag2']);
//                 tagAll.push(rows[step]['tag3']);
//                 tagAll.push(rows[step]['tag4']);
//                 tagAll.push(rows[step]['tag5']);
//                 tagAll.push(rows[step]['tag6']);
//                 tagAll.push(rows[step]['tag7']);
//                 filteredStandards.push(rows[step]);
//             }
//             }
//         var tagAllUnique = [];
//         resolve(tagAllUnique = Array.from(new Set(tagAll.filter(function (el) {
//             return el !== "";
//             }))));
//         // console.log('All tags for tag0 --->',tag0, tagAllUnique);
//         // console.log('All standards for these tag0 --->',tag0, filteredStandards);
//     });
// }
