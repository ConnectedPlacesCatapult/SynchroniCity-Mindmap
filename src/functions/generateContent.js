import { tag0Unique, tagAllUnique, filteredSubStandards, rows, getTags, filteredStandards, getFilteredSubStandards } from '../api/spreadsheet';
import { navigate } from '../components/Navbar/Navbar';
export let getId;
export let id;
export let idArray = [];
export let subId;
export let subIdArray = [];
export let value;
export let valueArray = [];
export let standard;
export let topic;
export let subtopic;
export let element;
let myElement;
let title;


const generateContent = {
    tag0: () => {
        return new Promise((resolve, reject) => {
            getId = (get) => { console.log(get) }
            for (let i = 0; i < tag0Unique.length; i++) {
                idArray.push(tag0Unique[i]);

                myElement = document.getElementById('tag0Container').innerHTML + '<div class="topicDiv" value="' + ( i + 1 ) + '"><img class="topicBG" src="' + process.env.PUBLIC_URL + '/Icon_Topics/png/Icon_Topic_' + ( i + 1 ) + '.png" /><p class="topic">' + tag0Unique[i] + '</p></div>'
                document.getElementById('tag0Container').innerHTML = myElement;
            }
            resolve('resolved');
        });
    },

    subtopics: () => {
        return new Promise((resolve, reject) => {
            subIdArray = []
            document.getElementById('topicList').innerHTML = '<li id="all">All</li>';
            for (let i = 0; i < tagAllUnique.length; i++) {
                subIdArray.push(tagAllUnique[i]);
                document.getElementById('topicList').innerHTML = document.getElementById('topicList').innerHTML + '<li id="' + subIdArray[i] + '" class="subtopic" value="' + i + '">' + tagAllUnique[i] + '</li>';
            }
            resolve('resolved');
        })
    },

    standards: () => {
        return new Promise((resolve, reject) => {
            document.getElementById('results').innerHTML = '';

            for (let i = 0; i < filteredStandards.length; i++) {
                title = filteredStandards[i].serial.replace(/[^\w\s]|\s/g, '').toLowerCase();
                document.getElementById('results').innerHTML = document.getElementById('results').innerHTML + '<div id=' + title + ' class="result" value="' + i +'"><p class="standard-serial">' + filteredStandards[i].serial + '</p><div class="abstract-container"><p>' + filteredStandards[i].abstract + '</p></div></div>';
            }
        })
    },

    standardsBySubtopic: () => {
        document.getElementById('results').innerHTML = '';

        for (let i = 0; i < filteredSubStandards.length; i++) {
            title = filteredSubStandards[i].serial.replace(/[^\w\s]|\s/g, '').toLowerCase();
            document.getElementById('results').innerHTML = document.getElementById('results').innerHTML + '<div id=' + title + ' class="result" value="' + i + '"><p class="standard-serial">' + filteredSubStandards[i].serial + '</p><div class="abstract-container"><p>' + filteredSubStandards[i].abstract + '</p></div></div>';
        }
    },

    assignTopicId: () => {
        return new Promise((resolve, reject) => {
            for (let i = 0; i < idArray.length; i++) {
                document.querySelector('.topicDiv:nth-child(' + (i + 1) + ')').id = idArray[i];
                document.querySelector('.topicDiv:nth-child(' + (i + 1) + ')').onclick = () => { subtopic = idArray[i]; getTags(rows, subtopic); generateContent.standards(); generateContent.crossRef(); generateContent.subtopics(); generateContent.assignSubtopicId(); generateContent.assignStandardId(); navigate.subtopic(); topic = tag0Unique[i]; document.getElementById('chosenTopic').innerHTML = topic; element = document.getElementById(idArray[i]).innerHTML; console.log(element); document.getElementById('putElement').innerHTML = element; };
            }

            resolve('resolved');
        })
    },

    assignSubtopicId: () => {
        return new Promise((resolve, reject) => {
            for (let i = 0; i < subIdArray.length; i++) {
                document.getElementById(subIdArray[i]).onclick = () => { subId = subIdArray[i]; getFilteredSubStandards(filteredStandards, subIdArray, subId); generateContent.standardsBySubtopic(); generateContent.assignStandardId(); }
            }
        });
    },

    assignStandardId: () => {
        return new Promise ((resolve, reject) => {
            for (let i = 0; i < filteredSubStandards.length; i++) {
                document.getElementById(filteredSubStandards[i].serial.replace(/[^\w\s]|\s/g, '').toLowerCase()).onclick = () => { generateContent.createModal(i); }
            }
        });
    },

    crossRef: () => {
        for (let i = 0; i < filteredStandards.length; i++) {
            document.querySelector('.result:nth-child(' + (i + 1) + ')').onclick = () => { generateContent.createModal(i) };
        }
    },

    createModal: (value) => {
        return new Promise((resolve, reject) => {
            standard = filteredStandards[value];
            console.log(standard);

            document.getElementById('resultWrapper').style.display = 'flex';
            setTimeout(() => { document.getElementById('standardsResult').style.opacity = '1'; }, 200);

            document.getElementById('standardName').innerHTML = standard.serial;
            document.getElementById('subTitle').innerHTML = standard.title_full;
            if (standard.abstract === "") {
                document.getElementById('description').innerHTML = "<strong>Sorry, there is not a description available at the moment.</strong>"
            } else {
            document.getElementById('description').innerHTML = standard.abstract;
            }
            document.getElementById('URL').setAttribute("href", standard.url);
        });
    },

    closeModal: () => {
        document.getElementById('standardsResult').style.opacity = '0';
        setTimeout(() => {document.getElementById('resultWrapper').style.display = 'none'}, 1000);
    }
}

export default generateContent;