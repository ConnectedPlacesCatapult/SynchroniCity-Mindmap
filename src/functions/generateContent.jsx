import { tag0Unique, tagAllUnique, filteredSubStandards, rows, getTags, filteredStandards } from '../api/spreadsheet';
import aesthetics from './aesthetics';
export let value;
export let getId;
export let id;
export let idArray = [];
export let standard;
export let subtopic;
let myElement;
let title;


const generateContent = {
    tag0: (imgPath) => {
        return new Promise((resolve, reject) => {
            getId = (get) => { console.log(get) }
            for (let i = 0; i < tag0Unique.length; i++) {
                idArray.push(tag0Unique[i]);

                myElement = document.getElementById('tag0Container').innerHTML + '<div class="topicDiv"><img class="folder" src="/static/media/folder_red.90acdd63.png" alt="Topic Icon"/><p class="topic">' + tag0Unique[i] + '<p></div>'
                document.getElementById('tag0Container').innerHTML = myElement;
            }

            resolve('resolved');
        });
    },

    subtopics: () => {
        return new Promise((resolve, reject) => {
            for (let i = 0; i < tagAllUnique.length; i++) {
                document.getElementById('subtopicsContainer').innerHTML = document.getElementById('subtopicsContainer').innerHTML + '<div class="topicDiv"><img class="folder" src="/static/media/folder_red.90acdd63.png" alt="Topic Icon"/><p class="topic">' + tagAllUnique[i] + '<p></div>'
            }
            resolve(console.log());
        })
    },

    standards: () => {
        return new Promise((resolve, reject) => {
            document.getElementById('results').innerHTML = ''

            for (let i = 0; i < filteredStandards.length; i++) {
                title = filteredStandards[i].serial.replace(/[^\w\s]|\s/g, '').toLowerCase();
                document.getElementById('results').innerHTML = document.getElementById('results').innerHTML + '<div id=' + title + ' class="result"><p class="standard-serial">' + filteredStandards[i].serial + '</p><div class="abstract-container"><p>' + filteredStandards[i].abstract + '</p></div></div>';
                document.getElementById(title).onclick = () => { console.log('onclickk') }
            }

            // for(let i = 0; i < filteredStandards.length; i++) {
            //     if(standardName === filteredStandards[i].serial) {
            //         resolve(standard = filteredSubStandards[i]);
            //         console.log(standard);
            //     } else if (i === filteredSubStandards.length) {
            //         reject(console.log('error'));
            //     }
            // }
        })
    },

    assignId: () => {
        for (let i = 0; i < idArray.length; i++) {
            document.querySelector('.topicDiv:nth-child(' + (i + 1) + ')').id = idArray[i];
            document.querySelector('.topicDiv:nth-child(' + (i + 1) + ')').onclick = () => { subtopic = idArray[i]; console.log(subtopic); getTags(rows, subtopic); generateContent.standards(); };
        }


    }
}

export default generateContent;