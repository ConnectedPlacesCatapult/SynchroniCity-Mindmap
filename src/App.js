import React, { useState, useEffect } from 'react';
import Navbar from './components/Navbar/Navbar';
import Header from './components/Header/Header';
import Topics from './pages/Topics/Topics';
import Standards from './pages/Standards/Standards';
import generateContent, { value, getId, standard, subtopic } from './functions/generateContent';
import aesthetics from './functions/aesthetics';
import { importSheet, getTag0, getTags, rows, tag0Unique, tagAllUnique, filteredStandards, filteredSubStandards, getFilteredSubStandards } from './api/spreadsheet';
import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
import '../node_modules/bootstrap/dist/js/bootstrap';
import './App.css';

function App() {
  useEffect(() => {
    // tag0 = click button
    var tag0 = 'MIMs';
    getTag0().then(() => { getTags(rows, subtopic) })
      .then(() => { generateContent.tag0(); })
      .then(() => { generateContent.assignTopicId(); })
      .then(() => { console.log(tagAllUnique); });

  });
  // tag0 = click button topic
  var tag0Select = 'MIMs'
  // subtag = click button subtopic
  var subTagSelect = 'API'
  // getTag0().then(() => {getTags(rows, subtopic)}).then(() => {getFilteredSubStandards(filteredStandards, subTagSelect)}).then(() => console.log(filteredSubStandards));
  // tag0Unique stores all topics
  // tagAllUnique stores all subtopics based on a topic
  // filteredStandards stores all standards based on a topic
  // filteredSubStandard stores all standards based on a subtopic and topic

  return (
    <div className="App">
      <Header />
      <Navbar />
      <Topics />
      <Standards />
    </div>
  );
}

export default App;
