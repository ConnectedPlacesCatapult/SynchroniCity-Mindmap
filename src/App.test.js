import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow';
import { importSheet, getTag0, getTags, rows, tag0Unique, tagAllUnique, filteredStandards, filteredSubStandards, getFilteredSubStandards } from './api/spreadsheet';
import generateContent from './functions/generateContent';



it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
});

it('returns shallow App', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const result = renderer.getRenderOutput();
  expect(result.type).toBe('div');
});


expect.extend({
  toContainObject(received, argument) {
    const pass = this.equals(received, 
      expect.arrayContaining([
        expect.objectContaining(argument)
      ])
    )
    if (pass) {
      return {
        message: () => (`expected ${this.utils.printReceived(received)} not to contain object ${this.utils.printExpected(argument)}`),
        pass: true
      }
    } else {
        return {
        message: () => (`expected ${this.utils.printReceived(received)} to contain object ${this.utils.printExpected(argument)}`),
        pass: false
      }
    }
  }
})

const state = [
  { 0: "Internet of Things",
    1: "Public Realm",
    2: "MIMs",
    3: "Mobility",
    4: "Health",
    5: "Smart City Governance",
    6: "Smart City ICT",
    7: "Data Management & AI",
    8: "Buildings",
    9: "Citizen Centric Services",
    10: "Environment",
    11: "Utilities"},
]


test('check tag0Unique array contains correct objects', () => {
  expect(state).toContainObject({ 5: 'Smart City Governance' })
  expect(state).not.toContainObject({ 10: 'Environmenta' })
});