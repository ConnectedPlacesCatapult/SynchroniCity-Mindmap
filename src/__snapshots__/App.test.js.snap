// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    id="header"
  >
    <img
      alt="Placeholder Logo"
      id="headerLogo"
      src="placeholder_header_logo.png"
    />
    <ul>
      <li
        className="headerLink"
      >
        home
      </li>
      <li
        className="headerLink"
      >
        cities
      </li>
      <li
        className="headerLink"
      >
        about
      </li>
      <li
        className="headerLink"
      >
        news
      </li>
      <li
        className="headerLink"
      >
        events
      </li>
      <li
        className="headerLink"
      >
        contact
      </li>
      <li
        className="headerLink"
      >
        become a member
      </li>
      <div
        id="headerIcons"
      >
        <img
          alt="search"
          id="search"
          src="search.png"
        />
        <img
          alt="hamburger"
          id="hamburger"
          src="hamburger.png"
        />
      </div>
    </ul>
  </div>
  <div
    className="navbar"
    id="navbar"
  >
    <img
      alt="Synchronicity Logo"
      className="navbarLogo"
      onClick={[Function]}
      src="logo_animated3.gif"
    />
    <ul>
      <li
        id="navAbout"
        onClick={[Function]}
      >
        About
      </li>
      <li
        id="navGI"
        onClick={[Function]}
      >
        Get Involved
      </li>
      <li
        id="navTopic"
        onClick={[Function]}
      >
        Topic 
      </li>
      <li
        id="seperator"
      >
        |
      </li>
      <li
        id="navSubtopic"
        onClick={[Function]}
      >
        Subtopic
      </li>
    </ul>
  </div>
  <section
    id="section2"
  >
    <div
      className="row"
      id="topicsSection"
    >
      <div
        className="col-lg-2 col-md-2 col-sm-10 offset-lg-1 offset-md-1 offset-sm-1 topicDescriptionDiv"
        id="TopicsTopBoxContainer"
      >
        <h2
          className="topicsHeader"
        >
          Select topic
        </h2>
        <p
          id="topicsText"
        >
          All standards are categorised into these topics. Please select one to continue
        </p>
        <img
          id="DescriptionBottomBox"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "Description_Box_Bottom.svg",
            }
          }
        />
      </div>
      <div
        className="col-lg-8 col-md-8 col-sm-10"
        id="tag0Container"
      />
    </div>
    <div
      id="resultsSection"
    >
      <div
        id="subtopicNavigationContainer"
      >
        <div
          id="putElement"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-12 col-md-12 col-sm-12 center z"
            >
              <div
                id="chosenTopicImg"
              />
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-lg-4 col-md-6 col-sm-10 offset-lg-4 offset-md-3 offset-sm-1 center"
            >
              <h2
                id="chosenTopic"
              >
                Chosen Topic
              </h2>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-8 col-md-8 col-sm-10 offset-lg-2 offset-md-2 offset-sm-1 center"
          >
            <ul
              id="topicList"
            >
              <div
                className="all"
              />
            </ul>
          </div>
        </div>
      </div>
      <div
        className="resultsRow row"
        id="resultsRow"
      >
        <div
          className="col-lg-2 col-md-2 col-sm-12 offset-lg-1 offset-md-1"
        >
          <h2
            className="topicsHeader"
          >
            All Standards
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
          </p>
        </div>
        <div
          className="col-lg-8 col-md-8 col-sm-12"
        >
          <div
            id="results"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="exampleModalLabel"
        className="modal fade"
        id="resultContent"
        role="dialog"
        tabIndex="-1"
      >
        <div
          className="modal-dialog"
          role="document"
        >
          <div
            id="mymodal"
          >
            <img
              id="modalImg"
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "Card_Result_Active.svg",
                }
              }
            />
            <h1>
              MY MODAL
            </h1>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    id="resultWrapper"
  >
    <div
      className="standardsResult row"
      id="standardsResult"
    >
      <div
        id="backdrop"
        onClick={[Function]}
      />
      <div
        className="standardsContentContainer col-lg-6 col-md-6 col-sm-10 offset-lg-3 offset-md-3 offset-sm-1"
      >
        <img
          id="resultTop"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "Close_Container.svg",
            }
          }
        />
        <div
          id="closeStandard"
          onClick={[Function]}
        />
        <div
          id="titleAndSubtitle"
        >
          <h3
            id="standardName"
          >
             Standard
          </h3>
          <h3
            id="subTitle"
          >
             Subtitle
          </h3>
        </div>
        <p
          id="description"
        >
           Description
        </p>
        <div
          id="URLDiv"
        >
          <a
            href="#"
            id="URL"
            target="_blank"
          >
             link to standard
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    id="CTASection"
  >
    <div
      className="row CTARow"
    >
      <div
        className="col-lg-3 col-md-3 col-sm-10 offset-lg-1 offset-md-1 offset-sm-1"
        id="CTASectionBox"
      >
        <div
          id="CTASectionDescription"
        >
          <h2>
            Get involved
          </h2>
          <p>
            Work with us to help the adoption of standards for smart and sustainable cities and communities.
          </p>
        </div>
        <img
          id="DescriptionBottomBox"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "Description_Box_Bottom.svg",
            }
          }
        />
      </div>
      <div
        className="col-lg-7 col-md-7 col-sm-10 offset-lg-0 offset-md-0 offset-sm-1"
        id="GetInvolvedTopBoxContainer"
      >
        <img
          id="GetInvolvedTopBox"
          src="Top_Box_GetInvolved.png"
        />
        <div
          id="CTAContent"
        >
          <h4
            id="initialCTAHeader"
          >
            The tool aims to help match cities and technology service providers with standards.
          </h4>
          <p>
            Creating a more discoverable, open and collaborative standards marketplace.  Ultimately it aims to help adoption of standards by making it easier to find standards for smart cities.  It also aims to help standards bodies by making their content more accessible and helping coordinate standards development work.  You can get involved in the following ways:
          </p>
          <br />
          <p>
            <strong>
              Cities and communities:
            </strong>
             We want cities and communities to get involved by using the tool and getting in touch to share your feedback plus any stories of how you have used standards yourselves.  You can also get in touch with any questions you might have.
          </p>
          <br />
          <p>
            <strong>
              Standards bodies:
            </strong>
             We are looking to work with standards bodies who would like to further promote their standards for cities through this platform by helping work with us on how we develop this further and by contributing your content.
          </p>
          <br />
          <p>
            If you want to get involved, contact us at info@oascities.org.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    id="aboutSection"
  >
    <div
      className="row aboutRow"
    >
      <div
        className="col-lg-3 col-md-3 col-sm-10 offset-lg-1 offset-md-1 offset-sm-1"
        id="AboutSectionBox"
      >
        <div
          id="aboutSectionDescription"
        >
          <h2>
            About the tool
          </h2>
          <p>
            A new tool to help match cities and technology service providers with standards. Creating a more discoverable, open and collaborative standards marketplace.
          </p>
        </div>
        <img
          id="DescriptionBottomBox"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "Description_Box_Bottom.svg",
            }
          }
        />
      </div>
      <div
        className="col-lg-7 col-md-7 col-sm-10 offset-lg-0 offset-md-0 offset-sm-0"
        id="AboutTopBoxContainer"
      >
        <img
          id="AboutTopBox"
          src="Top_Box_About.png"
        />
        <div
          id="aboutContent"
        >
          <h4
            id="initialHeader"
          >
            The Synchronicity standards catalogue is a prototype tool which aims to help match cities and technology service providers with standards.
          </h4>
          <p>
            It does this by bringing together standards for Smart Cities from many organisations into one place to make it easier to identify what standards to use, to create a more discoverable, open and collaborative standards marketplace.
          </p>
          <br />
          <p>
            For Synchronicity, this helps you discover the standards that enable minimum interoperability (Minimum Interoperability Mechanisms), but also helps discover other standards that help with you with your journey ultimately enabling smart, sustainable cities and communities.  This includes for example:
          </p>
          <br />
          <ul>
            <li>
              Guidance on what it means to be a smart city (see Smart City Governance &gt; Guidance)
            </li>
            <li>
              Knowing which standards can help you best think about managing data or how to use AI (see Data Management & AI category)
            </li>
            <li>
              What standards might help you when designing and procuring new street lighting within a public realm project (see Public Realm &gt; Street Lighting) or ICT systems and applications (see Smart City ICT)
            </li>
          </ul>
          <br />
          <p>
            The tool allows you to choose from a set of 12 core topics, from standards that help city governance, data management and citizen service design through to standards that help with smart mobility, utilities and health. For example within the data governance and AI category, you can focus in on standards relating to data privacy. You will then be presented with key information about the standards that can help including an abstract and a link to where the standard can be obtained.
          </p>
          <p>
            In the future, we would also like to evolve the tool to provide further functionality. Such as allowing cities to see use cases, to learn from others that have used the standard and data on how popular the standard is.  For standards bodies, we would like this to be a tool to help see where there are clashes between standard or gaps that exist, to help coordination and development.
          </p>
          <h4
            className="aboutHeader"
          >
            Vision:
          </h4>
          <p>
            Standards are the building blocks of the digital single marketplace.
          </p>
          <ul>
            <li>
              For the supply-side, they enable reusability, replicability and scalability.
            </li>
            <li>
              For the demand side, they enable replaceability, portability and comparability.
            </li>
          </ul>
          <p>
            For all parties, they enable interoperability, reducing risk, increasing investment and ultimately driving innovation.
          </p>
          <p>
            Yet with thousands of standards for smart cities, how does a city or technology service provide decide which standards to use?  And for standards bodies, how do they coordinate their work to ensure they are not repeating work, further contributing to a fragmented standards marketplace.
          </p>
          <p>
            Imagine if you could discover all global smart city standards in one single, user friendly platform, compare them, see who is using them and learn from other peoples experiences.  The vision for the standards library is to help create a more open, user friendly, interoperable standards marketplace, by providing a platform for standards organisations to collaborate and for cities and technology services providers to easily discover, decide upon and share what standards they are using.
          </p>
          <h4
            className="aboutHeader"
          >
            Why?
          </h4>
          <p>
            User research conducted by Connected Places Catapult and OASC identified that cities have a challenge with navigating the complex standards landscape, with X% of cities surveyed not knowing what standards to use.  Standards Development Organisations also have a challenge in coordinating work, often developing conflicting standards e.g. there are multiple IoT reference architectures.  This challenge has led to the development of a new 
            <strong>
              joint task force between ITU, ISO and IEC
            </strong>
            , the three main international standards bodies in order to better coordinate standards.
          </p>
          <p>
            The vision for the  online standards navigation tool is to help bring together the multiple standards available for smart cities and present this in an easy to navigate way for cities and technology service providers, so they can find what standards are relevant to what they are working on and see who else is using them, to help decide which standards to use.  For standards bodies, the vision is that this is a tool to enable collaboration by better coordination of standards development, helping to see where there are gaps as well as a tool to help the uptake of standards.
          </p>
          <br />
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
    id="footer"
  >
    <div
      className="col-lg-4 col-md-4 col-sm-4 center"
    >
      <img
        alt="Placeholder Logo"
        id="footerLogo"
        src="placeholder_header_logo.png"
      />
    </div>
    <div
      className="col-lg-4 col-md-4 col-sm-4 footerMiddleCol"
    >
      <h3>
        newsletter
      </h3>
      <p>
        Subscribe to our mailing list
      </p>
      <input
        type="text"
      />
      <button
        id="footerButton"
      >
        subscribe
      </button>
    </div>
    <div
      className="col-lg-4 col-md-4 col-sm-4 center"
    >
      <div
        id="contain"
      >
        <img
          className="footerPlaceholderImage"
          src="placeholder_image_footer.png"
        />
        <img
          className="footerPlaceholderImage"
          src="placeholder_image_footer.png"
        />
        <img
          className="footerPlaceholderImage"
          src="placeholder_image_footer.png"
        />
        <img
          className="footerPlaceholderImage"
          src="placeholder_image_footer.png"
        />
        <img
          className="footerPlaceholderImage"
          src="placeholder_image_footer.png"
        />
        <img
          className="footerPlaceholderImage"
          src="placeholder_image_footer.png"
        />
      </div>
    </div>
  </div>
</div>
`;
